### PR TITLE
feat(nav-bar-content): create RequirementViewTitle component

### PR DESCRIPTION
**Pull request by lisli1**
#### Description of changes

Created `RequirementViewTitle` component to be used in `RequirementView` from #2664.

#### Pull request checklist

- [x] Partially addresses an existing issue: # 1715693
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/tests/unit/tests/DetailsView/components/requirement-view-title.test.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import {
+    RequirementViewTitle,
+    RequirementViewTitleProps,
+} from 'DetailsView/components/requirement-view-title';
+import { shallow } from 'enzyme';
+import * as React from 'react';
+
+describe('RequirementViewTitleTest', () => {
+    it('renders with content from props', () => {
+        const props = {
```
Comment (JGibson2019) [inline src/tests/unit/tests/DetailsView/components/requirement-view-title.test.tsx:12]: Should you add some stubs for guidance links?

File: src/DetailsView/components/requirement-view-title.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    name: string;
+    guidanceLinks: HyperlinkDefinition[];
+    infoAndExamples: ContentPageComponent;
+}
+
+export const RequirementViewTitle = NamedFC<RequirementViewTitleProps>(
+    'RequirementViewTitle',
+    props => {
+        return (
+            <h1 className="requirement-view-title">
```
Comment (smoralesd) [inline src/DetailsView/components/requirement-view-title.tsx:22]: should this class come from a scss file?

File: src/tests/unit/tests/DetailsView/components/requirement-view-title.test.tsx (hunk 1 above)
Comment (waabid) [reply src/tests/unit/tests/DetailsView/components/requirement-view-title.test.tsx:12]: any props that are being used should be here and stubbed out.

File: src/DetailsView/components/requirement-view-title.tsx (hunk 2 above)
Comment (lisli1) [reply src/DetailsView/components/requirement-view-title.tsx]: Added requirement-view-title.scss

File: src/tests/unit/tests/DetailsView/components/requirement-view-title.test.tsx (hunk 1 above)
Comment (lisli1) [reply src/tests/unit/tests/DetailsView/components/requirement-view-title.test.tsx]: Added props stubs

File: src/DetailsView/components/requirement-view.tsx [hunk 3]
```diff
@@ -2,12 +2,28 @@
...
 export const RequirementView = NamedFC<RequirementViewProps>('RequirementView', props => {
-    return <div>{props.requirement.description}</div>;
+    return (
+        <div>
+            <RequirementViewTitle
+                deps={props.deps}
+                name={props.requirement.name}
+                guidanceLinks={props.requirement.guidanceLinks}
+                infoAndExamples={props.requirement.infoAndExamples}
+            ></RequirementViewTitle>
```
Comment (waabid) [inline src/DetailsView/components/requirement-view.tsx:25]:
```suggestion
            />
```

**Pull request merged**